### DISCUSSION TITLE
Installation dependency issue when running briefs to build Android

### DISCUSSION THREAD

**Discussion by lcgjhanfeng** [beeware/beeware#420] [Q&A] [answered]
hi,
I'm not sure if it's a problem with Beeware

but I added OpenCV Python and NumPy modules when using Python. However, when I run the briefcase build Android 
When it comes to https://pypi.org/simple ,  https://chaquo.com/pypi-13.1

These two addresses were searched for installation, but my access to them is very slow, which can cause a timeout error.

I have replaced the [global] index url in Python, but it still doesn't work when I build it https://pypi.org/simple ,  https://chaquo.com/pypi-13.1 . 

How can I change them?

**Comment by rmartin16** [comment]
These are currently hardcoded in to Briefcase; they cannot be changed. But if Briefcase is trying to download from them, that means you would need those packages to successfully build your app. There are not versions of opencv or numpy for Android available elsewhere.

What may work is attempting to manually download the wheel for these packages from https://chaquo.com/pypi-13.1 and specifying their file paths directly in to your `pyproject.toml`.

**Reply by lcgjhanfeng** [reply]
Thank you for your answer. 

My problem has been resolved and after a night of downloading, it has been cached, which improves the speed of each build. 

As for the reason for the error, I used Python 11. When I was building, I was reminded to use Python 3.8, but I didn't pay attention. When I switched to 3.8, it was already running normally. 

By the way, do you have a demo of how to write the file paths of the module to pyproject.toml? I didn't see any content related to this in the documentation, so I don't know how to write it.

**Reply by freakboy3742** [reply, marked as answer]
It's not documented because there's no real secret... you put a path in the requires setting:

```
requires = [
    "package-on-pypi==1.2.3",
    "/path/to/local/package",
    "./local/package.whl",
]
```
